diff --git i/lisp/emacs-lisp/comp.el w/lisp/emacs-lisp/comp.el
index 2bc8b996622..e66155cad7b 100644
--- i/lisp/emacs-lisp/comp.el
+++ w/lisp/emacs-lisp/comp.el
@@ -90,7 +90,8 @@ native-comp-bootstrap-deny-list
   :type '(repeat regexp)
   :version "28.1")
 
-(defcustom native-comp-compiler-options nil
+(defcustom native-comp-compiler-options
+  '(@compilerOptionsFlags@)
   "Command line options passed verbatim to GCC compiler.
 Note that not all options are meaningful and some options might even
 break your Emacs.  Use at your own risk.
@@ -101,8 +110,11 @@ native-comp-compiler-options
   :version "28.1")
 
 (defcustom native-comp-driver-options
-  (cond ((eq system-type 'darwin) '("-Wl,-w"))
-        ((eq system-type 'cygwin) '("-Wl,-dynamicbase")))
+  (append
+   (cond ((eq system-type 'darwin) '("-Wl,-w"))
+         ((eq system-type 'cygwin) '("-Wl,-dynamicbase")))
+   '(@backendPath@)
+   '(@driverOptionsFlags@))
   "Options passed verbatim to the native compiler's back-end driver.
 Note that not all options are meaningful; typically only the options
 affecting the assembler and linker are likely to be useful.
